image/qoi.go

```diff
@@ -48,6 +48,30 @@ func NewQOI[T pixel.Color](data string) (*QOI[T], error) {
 		// The colorspace affects how a pixel.Color is encoded.
 		return nil, errQOIUnsupportedColorSpace
 	}
+	img := &QOI[T]{
+		data: []byte(data),
```

Comment:
This will result in a heap allocation the size of the image, which is probably not what you want.
We could unsafely cast the string to a byte slice using `unsafe.Slice` (and then make sure to never modify it) to avoid this. It's a bit dangerous though.